src/resolve-info.ts

```diff
@@ -57,9 +67,13 @@ export interface TypeExpansion {
   [fieldName: string]: FieldExpansion | LeafField;
 }
 
-function createLeafField<T extends object>(props: T): T & LeafField {
+function createLeafField<T extends object>(
+  props: T,
+  shouldInclude: (__context: ShouldIncludeContext) => boolean
+): T & LeafField {
   return {
     [LeafFieldSymbol]: true,
+    __shouldInclude: shouldInclude,
```

Comment:
this function can be moved to the object's prototype instead of property, so it does not appear when iterating over the keys of the object.

Reply:
moved to prototypes and updated snapshots, might be better